run_unittests.py

```diff
@@ -3270,13 +3270,23 @@ def test_pkgconfig_gen_deps(self):
         self.assertEqual(sorted(out), sorted(['libfoo >= 1.0']))
 
         out = self._run(cmd + ['--cflags-only-other']).strip().split()
+        if is_osx():
+            # macOS does not use -pthread so add it in to make
+            # the test logic a bit easier to implement.
+            out += ['-pthread']
```

Comment:
I can see this going to create a very confusing output on failures suggesting there is `-pthread` present even though it is not by meson but from the test logic itself.

Reply:
I like the approach in https://github.com/mesonbuild/meson/pull/4305 better: `[] if is_osx() else ['-pthread']`.

Reply:
I think this needs to be `if_osx() or if_haiku()`, right?